### ISSUE TITLE
Can't use modifier keys for macros

### ISSUE THREAD

**Issue by Rollinc2198** [libratbag/piper#607]
**Information**

- `ratbagd` version (`ratbagd --version`): 0.14
- Piper version: 0.5.1-1
- Distribution: Ubuntu 20.10
- Device: Logitech G Pro Wireless
- Kernel version (ex. `uname -srmo`): Linux 5.8.0-33-generic x86_64 GNU/Linux

**Describe the bug**
When trying to set a macro to be *only* a key modifier (shift, ctrl, alt), it fails with the error message "Something went wrong. The device has been reset to a previous state." I can set other keybinds normally, and I can use these key modifiers in conjunction with other buttons.

**To Reproduce**
Please make sure your bug is reproducible with the CLI client, `ratbagctl`. If it is not open a bug report in libratbag instead.

1. Open Piper
2. Go to Buttons
3. Select button to modify
4. Scroll to bottom and select Send Keystroke
5. Press shift (or ctrl or alt) and hit Apply
6. Apply again and see error message

**Logs**
Please start the daemon with verbosity and reproduce the issue.

First make sure it isn't running already.
```
$ ratbagd --verbose=raw
ratbag error: Error while writing macro for button 4
ratbag error: hidpp20: failed to update button (-22)
ratbagd error: error committing device (-1000)
```

**Comment by epassaro** [comment]
Same error here.

**Information**

- `ratbagd` version (`ratbagd --version`): `0.15`
- Piper version:  `0.5.1`
- Distribution: `openSUSE Tumbleweed`
- Device: `Logitech M585`
- Kernel version (ex. `uname -srmo`): `Linux 5.9.14-1-default x86_64 GNU/Linux`

I'm not running `piper` with `sudo`, instead added user to `games` group.

**Comment by andrewbenton** [comment]
I have the same problem.  As an interesting point, my `G700s` that has working modifier keys - configured with the logitech utility on Windows a few years ago - sets the macro as a sequence of `<key> down, "Reserved" updown, <key> up`, but when setting new entries with Piper, it only shows as `<key> updown`.

**Information**

* `ratbagd` version (`ratbagd --version`): `0.14`
* Piper version: `0.5.1`
* Distribution: `Fedora 33`
* Device: `Logitech G903` and `Logitech G700s`
* Kernel version (ex. `uname -srmo`): `Linux 5.9.13-200.fc33.x86_64 x86_64 GNU/Linux`

**Comment by Mushoz** [comment]
Same error here! Is there anything we can do to help debug? Having modifier keys on the mouse would be VERY convenient if possible. Thanks!

**Comment by nYrXvcpk2l47ciid** [comment]
It would be really useful if either the sensitivity switch button e.g. on the G502 worked out of the box or if it could be bound to toggle a key like Alt, i.e. pressing the button presses Alt, releasing it releases Alt. Currently you can attempt to bind a key sequence but it doesn't appear to be possible to specify this functionality. Specifying the sequence as press-and-release results in the configuration not being able to be applied and it being reverted.

**Comment by shawngmc** [comment]
Also running into this issue on a G604. Trying to make a side button hold down shift, which would allow me to have it be a bar modification in WoW. Is anyone looking at this?

**Comment by Tickle** [comment]
Yeah, same issue here. Really darn annoying, trying to set up my hotkeys in UO and couldn't figure out why I wasn't able to set the key, esp when using modifier keys in a macro works just fine

**Comment by MattedPlace** [comment]
still an issue, had to use a key-remapper (kanata) as an alternate solution.

**Comment by Leshoyadut** [comment]
I'm running into this issue right now, myself. I used one of my thumb keys on my g502 as control and I can't seem to make it work on Piper at all. Super annoying since I liked having that a lot.

**Comment by Louis-Roberts** [comment]
I'm having the same issue. Strange thing is that the older G502SE that I just replace doesn't seem to have this problem. Only the newer G502 HERO that I picked up.

**Comment by tomasmiguez** [comment]
Same issue here on g pro hero.

**Comment by starcestus** [comment]
G502 lightspeed wireless getting the same issue.

**Comment by aaronpim** [comment]
I am also experiencing the same issue with G502 Hero wired.

**Comment by ajahstudio** [comment]
Ave got a bypass 

click ctrl, hold it and then click c then leave both keys at the same time it should appear as  `ctrl + c + ctrl`

save and try if it works for you. i dont know why but it worked on my G502 Wired.
